common/buildinfo.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2016 Pani Networks
+// All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+// Package common build information support.
+package common
+
+import (
+	"fmt"
+)
+
+// Build Information and Timestamp.
+// Pass build information to the executable using go run as below:
+//
+// go run -ldflags \
```

Comment:
it will print: "No Build Information Provided" if these flags are not there, i have already integrated them in jenkins (which i will update again once this commit hits master), also the makefile uses this here: https://github.com/romana/core/blob/testing/Makefile#L11